examples/concurrent_executions/concurrent_executions.c

```diff
@@ -0,0 +1,178 @@
+  relinquishment in perpetuity of all present and future rights to this
+  software under copyright law.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+  EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+  MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+  IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+  OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+  ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+  OTHER DEALINGS IN THE SOFTWARE.
+
+  For more information, please refer to <http://unlicense.org/>
+*/
+
+#include <stdio.h>
+#include <string.h>
+
+#include "cassandra.h"
+
+#define CONCURRENCY_LEVEL 32
+#define NUM_REQUESTS 10000
+
+CassSession* session = NULL;
+const CassPrepared* prepared = NULL;
+CassUuidGen* uuid_gen = NULL;
+
+int num_requests = 0;
+int num_outstanding_requests = 0;
+
+void print_error(CassFuture* future) {
+  const char* message;
+  size_t message_length;
+  cass_future_error_message(future, &message, &message_length);
+  fprintf(stderr, "Error: %.*s\n", (int)message_length, message);
+}
+
+CassCluster* create_cluster(const char* hosts) {
+  CassCluster* cluster = cass_cluster_new();
+  cass_cluster_set_contact_points(cluster, hosts);
+  return cluster;
+}
+
+CassError connect_session(CassSession* session, const CassCluster* cluster) {
+  CassError rc = CASS_OK;
+  CassFuture* future = cass_session_connect(session, cluster);
+
+  cass_future_wait(future);
+  rc = cass_future_error_code(future);
+  if (rc != CASS_OK) {
+    print_error(future);
+  }
+  cass_future_free(future);
+
+  return rc;
+}
+
+CassError execute_query(CassSession* session, const char* query) {
+  CassError rc = CASS_OK;
+  CassStatement* statement = cass_statement_new(query, 0);
+
+  CassFuture* future = cass_session_execute(session, statement);
+  cass_future_wait(future);
+
+  rc = cass_future_error_code(future);
+  if (rc != CASS_OK) {
+    print_error(future);
+  }
+
+  cass_statement_free(statement);
+  cass_future_free(future);
+  return rc;
+}
+
+CassError prepare_insert(CassSession* session, const CassPrepared** prepared) {
+  CassError rc = CASS_OK;
+  const char* query = "INSERT INTO examples.concurrent_executions (id, value) VALUES (?, ?);";
+
+  CassFuture* future = cass_session_prepare(session, query);
+  cass_future_wait(future);
+
+  rc = cass_future_error_code(future);
+  if (rc != CASS_OK) {
+    print_error(future);
+  } else {
+    *prepared = cass_future_get_prepared(future);
+  }
+
+  cass_future_free(future);
+
+  return rc;
+}
+
+void insert_into_concurrent_executions() {
+  CassFuture* futures[CONCURRENCY_LEVEL];
+  int num_requests = NUM_REQUESTS;
+
+  while (num_requests > 0) {
+    int i;
+    int num_outstanding_requests = CONCURRENCY_LEVEL;
+    if (num_requests < num_outstanding_requests) {
+      num_outstanding_requests = num_requests;
+    }
+
```

Comment:
Might be clearer if `num_requests` was updated here instead of below in the future check loop.

```c
num_request -= num_outstanding_requests;
```